sentry-types/src/protocol/envelope.rs

```diff
@@ -504,19 +522,17 @@ impl Envelope {
         Self::from_bytes_raw(bytes)
     }
 
-    fn parse_header(slice: &[u8]) -> Result<(EnvelopeHeader, usize), EnvelopeError> {
-        let mut stream = serde_json::Deserializer::from_slice(slice).into_iter();
-
-        let header: EnvelopeHeader = match stream.next() {
-            None => return Err(EnvelopeError::MissingHeader),
-            Some(Err(error)) => return Err(EnvelopeError::InvalidHeader(error)),
-            Some(Ok(header)) => header,
-        };
+    fn parse_headers(slice: &[u8]) -> Result<(EnvelopeHeaders, usize), EnvelopeError> {
+        let mut lines = slice.lines();
+        let first_line = lines
+            .next()
+            .ok_or(EnvelopeError::MissingHeader)?
+            .map_err(|err| EnvelopeError::InvalidHeader(Box::new(err)))?;
```

Comment:
Let's go back to `split` rather than `lines`. I did not realize that `lines` performs allocations and that it returns IO errors. 

But, please use `split` rather than `splitn` – performance-wise, the two should be identical, and the `split` is likely easier to reason about.